internal/comet/comet_conn.go

```diff
@@ -0,0 +1,47 @@
+package comet
+
+import (
+	"context"
+	"fmt"
+	grpc_retry "github.com/grpc-ecosystem/go-grpc-middleware/retry"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/connectivity"
+	"google.golang.org/grpc/credentials/insecure"
+	"strings"
+	"time"
+)
+
+// NewCometGRPCConn creates a new gRPC client connection with retry interceptors and connectivity checks.
+func NewCometGRPCConn(endpoint string) (*grpc.ClientConn, error) {
```

Comment:
would maybe suggest going with the naming convention of "core" rather than "comet" for this and the package naming. Or just pkg naming `core.NewCometGRPCConn`